api/swagger.yml

```diff
@@ -1439,6 +1494,14 @@ components:
         - requestFee
         - requestNonce
         - token
+    PoolL2Transactions:
+      type: object
+      properties:
+        transactions:
+          type: array
+          description: List of pool l2 transactions
+          items:
+            $ref: '#/components/schemas/PoolL2Transaction'
```

Comment:
Missing pagination info, and validation info:

```yaml
pendingItems:
          $ref: '#/components/schemas/PendingItems'
# ...
required:
        - transactions
        - pendingItems
      additionalProperties: false
```

Please have a look on how paginated endpoints are implemented, all of them follow the same pattern.

Reply:
I saw the pagination, started to implement it, but faced some issues. In tx_pool as a primary key we use txId, but our pagination mechanism assumes that we use uint type of primary key. What should I do for this type? Paginate by timestamp?

Reply:
You're absolutley right @Mikelle this table was not designed to be used with pagination.

There are a couple of doable workarounds: 

- Add a new coulum `item_id` to the table to be able to mimic the behaviour of the other paginated resources. To do so it's necessary to create a new file `0002.sql` under `db/migrations`. Note that this would be the first time we actually use the migration mechanism introducing schema changes. If you choose to implement this, please strongly test that existing data is migrated propperly.
- Add a numeric counter within the query, something like `select row_number() over(), tx_id from tx_pool;`. Due to the temporal nature of the data in this table, maintaining a strict order on the items is not important and this could work